### PR TITLE
add urls to csp header for ads

### LINKED ISSUE #6774: Investigation: Identify CSP header changes for dotcom-bootstrap Canonical Ads
Canonical Ads are currently on test but the Content Security Policy (CSP) header configuration is stopping them from loading.

Locally the cspHeader is skipped, but this can be temporarily removed here: https://github.com/bbc/simorgh/blob/b04a7263a1422ccb060392caf05999a086dd3c2e/src/server/index.jsx#L100

Load the page with the follow query param: http://localhost.dev.bbc.co.uk:7080/pidgin?adUnit=/4817/bbccom.test.site.flash/news_homepage_int

Investigate the following to try and get ads to load consistently:

1. Add https://gn-web-assets.api.bbc.com to `canonicalNonLive` in the following file so the main GNL script will load. 
https://github.com/bbc/simorgh/blob/ad880d00ee55efe008117c0f66dafa8104f275c3/src/server/utilities/cspHeader/index.js
2. We have some questions around additions that were required:
- https://securepubads.g.doubleclick.net has been flagged as having high serverity insecurities by our CSP validator.
- We have to include `'unsafe-eval'` in our CSP, which isn't something we're particularly comfortable with.
3. After this the google ads will try to load and also be blocked by CSP.
4. Investigate the Google Ads recommendations and attempt to implement them https://developers.google.com/tag-manager/web/csp
5. Investigate if we can have an allowlist the same as WebCore https://github.com/bbc/web/blob/765d57a1d300126939cda8dc6dc009d50fa07584/packages/website/headers/src/content-security-policy.config.js (**This has been tried and didn't seem to be complete**).

**This investigation should generate a list of assets (urls for JS, etc) that the dotcom-bootrap JS loads in, so we can have a conversation with the GNL team regarding this.**
**We should pair on this issue**

**Screenshot**
image

**Testing notes**
[Tester to complete]

Dev insight: Updating unit tests should be sufficient. 

On localhost these CSP headers currently only apply on the production build.
`npm run build:test && npm run start` & visit localhost URL
reviewing developer console to see no more CSP errors exist.

Once the PR is merged, we should not have those errors on `https://www.test.bbc.com/mundo`

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by thekp**
Resolves #6774

**Overall change:**
_Add links to our CSP header required for the current Ads._

example url: to see locally: http://localhost:7080/mundo?adUnit=/4817/bbccom.test.site.flash/news_homepage_int

You will have to run `npm run build && npm run start` to see the Ads working locally with CSP

**Code changes:**

- Updated unit tests
- Added the following urls for ads to our csp header

connectSrc:
```
      'https://csi.gstatic.com',
      'https://experience.tinypass.com',
      'https://static.test.files.bbci.co.uk',
      'https://survey.effectivemeasure.net',
      'https://detect-survey.effectivemeasure.net',
      'https://collector.effectivemeasure.net/',
      'https://www.bbc.co.uk',
      'https://adservice.google.com', // ads
      'https://securepubads.g.doubleclick.net', // ads
      'https://pagead2.googlesyndication.com', // ads
      'https://tpc.googlesyndication.com', // ads
```

frameSrc:
```
      'https://bcp.crwdcntrl.net',
      'https://edigitalsurvey.com',
      'https://securepubads.g.doubleclick.net', // ads
      'https://tpc.googlesyndication.com', // ads
```

imgSrc:
```
      'https://collector.effectivemeasure.net',
      'https://csi.gstatic.com',
      'https://sb.scorecardresearch.com',
      'https://secure-us.imrworldwide.com',
```

scriptSrc:
```
      'https://ad.crwdcntrl.net',
      'https://adservice.google.co.uk',
      'https://adservice.google.com',
      'https://bbc.gscontxt.net',
      'https://bcp.crwdcntrl.net',
      'https://cdn.ampproject.org',
      'https://collector.effectivemeasure.net',
      'https://me-ssl.effectivemeasure.net',
      'https://privacy.crwdcntrl.net',
      'https://sb.scorecardresearch.com',
      'https://securepubads.g.doubleclick.net',
      'https://t.effectivemeasure.net',
      'https://tags.crwdcntrl.net',
      'https://tpc.googlesyndication.com',
      'https://gn-web-assets.api.bbc.com',
```

defaultSrc:
```
      'https://*.safeframe.googlesyndication.com',
      'https://tpc.googlesyndication.com',
```

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/server/utilities/cspHeader/index.js
Comment (sareh) [inline src/server/utilities/cspHeader/index.js]: This is already in the CSP above. We should likely just update the comment, to say it's for ads as well as STY includes
```suggestion
```

File: src/server/utilities/cspHeader/index.js
Comment (sareh) [inline src/server/utilities/cspHeader/index.js]: Suggesting this change since these lines might be reorganised in future
```suggestion
      'https://bcp.crwdcntrl.net', // ads
      'https://edigitalsurvey.com', // ads
```

File: src/server/utilities/cspHeader/index.js
Comment (sareh) [inline src/server/utilities/cspHeader/index.js]:
```suggestion
      'https://csi.gstatic.com', // ads
      'https://experience.tinypass.com', // ads
      'https://static.test.files.bbci.co.uk', // ads
      'https://survey.effectivemeasure.net', // ads
      'https://detect-survey.effectivemeasure.net', // ads
      'https://collector.effectivemeasure.net', // ads
```

File: src/server/utilities/cspHeader/index.js
Comment (sareh) [inline src/server/utilities/cspHeader/index.js]:
```suggestion
      'https://collector.effectivemeasure.net', // ads
      'https://csi.gstatic.com', // ads
      'https://sb.scorecardresearch.com', // ads
      'https://secure-us.imrworldwide.com', // ads
```

File: src/server/utilities/cspHeader/index.js
Comment (sareh) [inline src/server/utilities/cspHeader/index.js]: When running locally with `npm run build && npm run start` & visiting http://localhost:7080/afrique, I saw CSP header errors, so I've suggested these additions, which fix the script-src errors:
```suggestion
      'https://gn-web-assets.api.bbc.com', // ads
      'https://pixel.adsafeprotected.com', // ads
      'https://static.adsafeprotected.com', // ads
      'https://www.googletagservices.com', // ads
```

Comment (sareh) [reply src/server/utilities/cspHeader/index.js]: Additionally, to make the CSP errors not show in the console, the following need to be added to the 
imgSrc, canonicalNotLive list:
```
      'https://pixel.adsafeprotected.com', // ads
      'https://dt.adsafeprotected.com', // ads
```

**Review by sareh — COMMENTED** [review]
Thanks for the additions. I've added some comments below, regarding additions to the headers.


One thing to note is that although we're adding these to the non Live CSP headers just now, so we can see the adverts working on our Test environment, in a future issue we should review the chain of assets that ads are making requests to and work with the GNL team to see if we can reduce the number of requests being made.

**Review by sareh — APPROVED** [review]
When running `npm run build && npm run start`, I can see ads on these front pages: 
http://localhost:7080/afrique?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
http://localhost:7080/hindi?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
http://localhost:7080/mundo?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
http://localhost:7080/portuguese?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
http://localhost:7080/russian?adUnit=/4817/bbccom.test.site.flash/news_homepage_int

Screenshot leaderboard ad on Afrique

However I don't see ads on Arabic & Zhongwen
http://localhost:7080/arabic?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
http://localhost:7080/zhongwen/simp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
We can have a separate PR to fix the toggles for these.

File: src/server/utilities/cspHeader/index.test.js
Comment (tochwill) [inline src/server/utilities/cspHeader/index.test.js]:
```suggestion
        'https://collector.effectivemeasure.net',
```

File: src/server/utilities/cspHeader/index.js
Comment (tochwill) [inline src/server/utilities/cspHeader/index.js]:
```suggestion
      'https://collector.effectivemeasure.net', // ads
```

**Review by tochwill — COMMENTED** [review]
Looks good! Just one question, I've seen issues with trailing slashes in CSPs before, are they in there for a reason?

**Review by tochwill — APPROVED** [review]

**Comment by thekp** [comment]
Merge this PR after: https://github.com/bbc/mozart-routing/pull/1407

**Comment by PriyaKR** [comment]
On safari browser still i am seeing csp errors.Refer screenshot.
Safari_csp

**Pull request merged**
